agent/engine/docker_task_engine.go

```diff
@@ -106,8 +106,7 @@ func NewDockerTaskEngine(cfg *config.Config, client DockerClient, credentialsMan
 		managedTasks:  make(map[string]*managedTask),
 		taskStopGroup: utilsync.NewSequentialWaitGroup(),
 
-		containerEvents: make(chan api.ContainerStateChange),
-		taskEvents:      make(chan api.TaskStateChange),
+		stateChangeEvents: make(chan statechange.StateChangeEvent),
```

Comment:
I like @richardpen's suggestion of eventhandler owning this channel. It's fine  if you want to address this in another PR. But, we should do it sooner rather than later I think.